app/controllers/events_controller.rb

```diff
@@ -0,0 +1,34 @@
+class EventsController < ApplicationController
+  def index
+    @events = Event.all
+  end
+
+  def new
+    @event = current_user.events.build
+  end
+
+  def create
+    @event = current_user.events.build(event_params)
+    if @event.save
+      flash[:notice] = 'Event Created Successfully'
+      redirect_to root_path
+    else
+      flash.now[:alert] = 'Form not submitting for some reasons'
+      render :new
+    end
+  end
+
+  def show
+    @event = Event.find(params[:id])
+  end
```

Comment:
### You can list events based on the following criteria:-
lists events created by user `@created_events = current_user.events`
lists events previoulsy attendented by user `@past_events = current_user.attended_events.previous`
lists upcoming events previously attended by user `@upcoming_events = current_user.attended_events.upcoming`